### ISSUE TITLE
Scripts for computing IoU

### ISSUE THREAD

**Issue by rahul-nt** [zlkanata/DeepGlobe-Road-Extraction-Challenge#3]
Hi

Is the code for computing IoU and mIoU also included in the scripts, or does that have to be done separately from the resultant predictions in the _submits_ directory?

**Comment by zlkanata** [comment]
There is no script for computing IoU or mIoU in this repository. Computing IoU using numpy just like this.
`def iou(a, b):
	a //= 255
	b //= 255
	o = a+b
	i = (o==2).sum()
	u = (o>=1).sum()
	if u == 0:
		return 1.0
	return 1.0*i/u`
![image](https://user-images.githubusercontent.com/19966245/45463775-316fca80-b740-11e8-9bf7-e07ed9bc20d6.png)

**Comment by rahul-nt** [comment]
Thanks for the reply.

Another question: how did you do the validation if the validation masks are not available in the dataset?
(I had also seen the same question of yours on the DeepGlobe forum, but there was no reply there)

**Comment by zlkanata** [comment]
For your question, I did not validate my models in the final phase.

I tried several methods in the development phase, and found that D-LinkNet34 is better than the ensemble of LinkNet34 and Unet on the validation set.
Therefore, I trained D-LinkNet50, D-LinkNet101 and D-Unet in the final phase, and submitted the predictions on the test set only.

It would be more standard to use some kind of validation such as cross-validation. But due to the time limitation and no many training images, I did not partition some of the data as validation set.
